Solution1/Solution1.Tests/WeatherApi/WeatherControllerTests.cs

```diff
@@ -0,0 +1,100 @@
+﻿using BusinessLayer.Command;
+using BusinessLayer.Command.Abstract;
+using BusinessLayer.Configuration.Abstract;
+using BusinessLayer.DTOs;
+using BusinessLayer.Services.Abstract;
+using KellermanSoftware.CompareNetObjects;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using WeatherApi.Controllers;
+using Xunit;
+
+namespace Weather.Tests.WeatherApi
+{
+    public class WeatherControllerTests
+    {
+        private readonly Mock<IInvoker> _invokerMock;
+        private readonly Mock<IWeatherServiсe> _weatherServiceMock;
+        private readonly Mock<IConfig> _config;
+        private readonly WeatherController _weatherController;
+        private readonly string cityName = "Minsk";
+
+        public WeatherControllerTests()
+        {
+            _invokerMock = new Mock<IInvoker>();
+            _weatherServiceMock = new Mock<IWeatherServiсe>();
+            _config = new Mock<IConfig>();
+
+            _weatherController = new WeatherController(_weatherServiceMock.Object, _config.Object, _invokerMock.Object);
+        }
+
+        [Fact]
+        public async Task GetCurrentWeatherByCityName_EnterCityName_ReturnWeather()
+        {
+            // Arrange
+            var temp = 10;
+            var comment = "TestComment";
+
+            var weather = new WeatherDTO()
+            {
+                CityName = cityName,
+                Temp = temp,
+                Comment = comment
+            };
+
+            _invokerMock
+                .Setup(invoker => invoker.RunAsync(It.IsAny<CurrentWeatherCommand>(), It.Is<CancellationToken>(x => !x.IsCancellationRequested)))
+                .ReturnsAsync(weather);
+            
+            //Act
+            var result = (OkObjectResult)(await _weatherController.GetCurrentWeatherByCityNameAsync(cityName)).Result;
```

Comment:
it can throw exception if it's not OkObjectResult. You shouldn't preform extra actions in test, if the result should be exactly OkObjectResult you should check it using Assert

Reply:
+